From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 1 May 2018 20:18:54 -0400
Subject: [PATCH] EndermanAttackPlayerEvent

Allow control over whether or not an enderman aggros a player.

This allows you to override/extend the pumpkin/stare logic.

diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
index c256fa48d64776ae7a7b21d42fb87c0a8ca7f3bc..d03c0b4ea5fad793a7bedad5ae3be0b779dc458c 100644
--- a/src/main/java/net/minecraft/server/EntityEnderman.java
+++ b/src/main/java/net/minecraft/server/EntityEnderman.java
@@ -167,7 +167,15 @@ public class EntityEnderman extends EntityMonster implements IEntityAngerable {
         this.a((WorldServer) this.world, nbttagcompound);
     }
 
+    // Paper start - OBFHELPER - ok not really, but verify this on updates
     private boolean g(EntityHuman entityhuman) {
+        boolean shouldAttack = g_real(entityhuman);
+        com.destroystokyo.paper.event.entity.EndermanAttackPlayerEvent event = new com.destroystokyo.paper.event.entity.EndermanAttackPlayerEvent((org.bukkit.entity.Enderman) getBukkitEntity(), (org.bukkit.entity.Player) entityhuman.getBukkitEntity());
+        event.setCancelled(!shouldAttack);
+        return event.callEvent();
+    }
+    private boolean g_real(EntityHuman entityhuman) {
+        // Paper end
         ItemStack itemstack = (ItemStack) entityhuman.inventory.armor.get(3);
 
         if (itemstack.getItem() == Blocks.CARVED_PUMPKIN.getItem()) {
